mars/scheduler/operands/common.py

```diff
@@ -313,7 +328,8 @@ def free_data(self, state=OperandState.FREED):
         if state == OperandState.CANCELLED:
             can_be_freed = True
         else:
-            can_be_freed = self._graph_ref.check_operand_can_be_freed(self._succ_keys)
+            can_be_freed = all(graph_ref.check_operand_can_be_freed(self._succ_keys) for
```

Comment:
``check_operand_can_be_freed`` can return None indicating an indeterministic state in which the free step shall be delayed. This change will break fail-over in some scenario.